test/http/read.cpp

```diff
@@ -96,6 +151,20 @@ class read_test
                 break;
         }
         BEAST_EXPECT(n < limit);
+        for(n = 0; n < limit; ++n)
+        {
+            streambuf sb;
+            sb.commit(buffer_copy(
+                sb.prepare(len), buffer(s, len)));
+            test::fail_counter fc{n};
+            test::string_stream ss{ios_, s};
+            parser_v1<isRequest, fail_body, headers> p{fc};
+            error_code ec;
+            parse(ss, sb, p, ec);
+            if(! ec)
+                break;
+        }
+        BEAST_EXPECT(n < limit);
```

Comment:
You are proposing that if the parse completes successfully on the first iteration through the loop, that it should be considered a test failure?

The purpose of this loop is to generate an error on the n-1th I/O operation. And it keeps going until the parse is successful. This needs to happen in some bounded number of I/Os, and that's what the test is testing.

Thinking about your idea, a way to improve the test is to take the parsed result and convert it back to a string, then compare against the input to make sure that the result of the parse matches. I can open an issue for that. Note that the purpose of this code is to increase test coverage by exercising all the failure branches, which usually never get tested.